Lib/email/_header_value_parser.py

```diff
@@ -2802,8 +2804,12 @@ def _refold_parse_tree(parse_tree, *, policy):
             wrap_as_ew_blocked -= 1
             continue
         tstr = str(part)
-        if part.token_type == 'ptext' and set(tstr) & SPECIALS:
+        if part.token_type == 'ptext':
             # Encode if tstr contains special characters.
+            if not SPECIALSNL.isdisjoint(tstr):
+                want_encoding = True
+        elif not NLSET.isdisjoint(tstr):
+            # Encode if tstr contains newlines.
             want_encoding = True
```

Comment:
Then I suggest:
```suggestion
        if not want_encoding:
            if part.token_type == 'ptext':
                # Encode if tstr contains special characters.
                want_encoding = not SPECIALSNL.isdisjoint(tstr)
            else:
                # Encode if tstr contains newlines.
                want_encoding = not NLSET.isdisjoint(tstr)
```
No need to check for special characters if `want_encoding` is already true.